### PR TITLE
check the address family of peers in get_peers

### PR DESCRIPTION
**Pull request by ssiloti**

### CODE REVIEW

File: src/kademlia/dht_storage.cpp [hunk 1]
```diff
@@ -247,6 +247,10 @@ namespace
 					// peer list
 					if (noseed && iter->seed) continue;
 
+					// only include peers with the right address family
+					if (iter->addr.protocol().family() != protocol.family())
```
Comment (arvidn) [inline src/kademlia/dht_storage.cpp]: wouldn't you need to take the ``want`` argument into account here too?

**Comment by arvidn** [comment]
good catch!

File: src/kademlia/dht_storage.cpp (hunk 1 above)
Comment (ssiloti) [inline src/kademlia/dht_storage.cpp]: The `want` argument only applies to the `nodes`/`nodes6` keys. The `value` key always contains peers with the same address family as the request was received from.

Comment (arvidn) [inline src/kademlia/dht_storage.cpp]: oh, right. of course

**Pull request merged**